Sources/TuistGenerator/Generator/LinkGenerator.swift

```diff
@@ -519,7 +519,7 @@ final class LinkGenerator: LinkGenerating { // swiftlint:disable:this type_body_
             default:
                 return nil
             }
-        }.map { ("$BUILT_PRODUCTS_DIR/\($0)", "$BUILT_PRODUCTS_DIR/$FULL_PRODUCT_NAME/Macros/\($0)") }
+        }.map { ("$SYMROOT/$CONFIGURATION/\($0)", "$BUILT_PRODUCTS_DIR/$FULL_PRODUCT_NAME/Macros/\($0)") }
```

Comment:
I'm not 100% confident about this one. Here's the thing, we use the static iOS framework as a bundle for the macOS executable representing the macro. So when compiling both, we end up with something like this:

```bash
Project/
  Build/
    Products/
      Debug/
        PackageMacroMacro # <- macOS Executable
      Debug-iphonesimulator/
        PackageMacro.framework # <- iPhone Simulator framework
```
If we use `$BUILT_PRODUCTS_DIR` from the `PackageMacro` framework target, it points to the `Debug-iphonesimulator` where the executable doesn't exist. The only configuration of variables that I found that points to the directory where the executable exists is `$SYMROOT/$CONFIGURATION`